mesoscope/cellpack
# Feature/mesh

Problem
=======
This makes it so we can have a mesh compartment. #15 
NOTE: this looks like a big PR because I deleted the mgl_tools folder

Solution
========
The main big change is the creation of a Mesh store, which holds all the functionality around loading and manipulating a mesh. I am mostly wrapping [Trimesh.py](https://trimsh.org/trimesh.html) in that class. 

Other changes:

- Updated the inner grid functions to accurately sort the grid points into inside, surface, and outside points. Currently, `trimesh` and `raytrace` work. I also switched from a bounding box check to a query ball check to gather the possible inside points which speeds up the calculation. 
- I added a grid visualization to the simularium output that puts little colored spheres at the grid points. 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)
* This change requires a documentation update


Steps to Verify:
----------------
1. run `analyze -p spheresSST -np -r /cellpack/test-recipes/test_compartment.json -d 3`
2. drop both the resulting file and `membrane_1.obj` on the viewer to see the packing and grid points

Screenshots (optional):
-----------------------
<img width="784" alt="Screen Shot 2022-07-14 at 12 53 10 PM" src="https://user-images.githubusercontent.com/5170636/179071545-9bb3fda1-cfdb-4be3-bd6b-6a23bfb4caef.png">
